fdb-relational-core/src/test/java/com/apple/foundationdb/relational/recordlayer/RelationalArrayTest.java

```diff
@@ -123,12 +124,9 @@ private void insertArraysViaQuerySimpleStatement() throws SQLException {
                         "['11', '22'], null, " +
                         "[x'31', x'32'], null, " +
                         "[(11, '11'), (22, '22')], null " +
-                        ")"));
-        // This should not go through since non-null array is not initialized, However, that is being currently getting
-        // initialized by an empty array (no elements in unwrapped REPEATED field in proto).
-        // This phenomenon is also true with basic types, i.e., non-nullability is not enforced and is tracked by
-        // TODO (Add support + tests for column nullable/not null)
-        insertQuery("INSERT INTO T (pk) VALUES (4)");
+                        ")")).hasErrorCode(ErrorCode.INTERNAL_ERROR);
```

Comment:
This is throwing INTERNAL_ERROR currently because the check for non-nullability is currently not happening during the record construction. It directly happens during execution which throws verifyException. Opened an issue about it https://github.com/FoundationDB/fdb-record-layer/issues/3133